app/models/git_repository.rb

```diff
@@ -25,6 +25,27 @@ def self.delete_repo_dir(id, directory_name)
     FileUtils.rm_rf(directory_name)
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  # We use this approach for now over Rugged#check_connection because of a segfault:
+  #
+  #   https://github.com/libgit2/rugged/issues/859
+  #
+  def check_connection?
+    require 'net/ping/external'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
+    url = url.sub(':', '/').sub('git@', 'https://') if url.start_with?('git@')
```

Comment:
I think we want to pass in `url`.

Reply:
`url` is actually a property of GitRepository. Maybe I should use a different variable name there, though, to avoid confusion.

Reply:
```
 Q-task_id([r18_service_template_provision_task_18]) MiqAeServiceModelBase.ar_method raised: <NoMethodError>: <undefined method `start_with?' for nil:NilClass>
log/automation.log:[----] E, [2020-10-30T15:53:23.520583 #49724:2adac77d199c] ERROR -- : Q-task_id([r18_service_template_provision_task_18]) /var/www/miq/vmdb/app/models/git_repository.rb:29:in `check_connection?'
log/automation.log:[----] E, [2020-10-30T15:53:23.522653 #49724:2adacab48d98] ERROR -- : Q-task_id([r18_service_template_provision_task_18]) <AEMethod check_connection> The following error occurred during method evaluation:
log/automation.log:[----] E, [2020-10-30T15:53:23.523028 #49724:2adacab48d98] ERROR -- : Q-task_id([r18_service_template_provision_task_18]) <AEMethod check_connection>   NoMethodError: undefined method `start_with?' for nil:NilClass
```

Reply:
@d-m-u updated

Reply:
Updated...again!

Reply:
Marking as resolved cause your update fixed the error in my previous comment (on a provision without the connectivity failing, will test again with the scenario from the ticket).